client/src/app/components/rsvp/rsvp.component.html

```diff
@@ -1,5 +1,12 @@
-<app-rsvp-form
-	[pokemon]="pokemon"
-	[existingRsvpData]="existingRsvpData"
-	(submit)="sendForm($event)"
-></app-rsvp-form>
+<div *ngIf="!pokemonValidated">
+	<app-code-input-form
+		(submit)="validatePokemon($event)"
+	></app-code-input-form>
+</div>
+<div *ngIf="pokemonValidated">
```

Comment:
Could be fun to take a look at 
https://angular.io/api/common/NgIf#showing-an-alternative-template-using-else

Not a mandatory change